pkg/labels/regexp.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package labels
+
+import (
+	"regexp"
+	"regexp/syntax"
+	"strings"
+)
+
+type FastRegexMatcher struct {
+	re     *regexp.Regexp
+	prefix string
+	suffix string
+}
+
+func NewFastRegexMatcher(v string) (*FastRegexMatcher, error) {
+	re, err := regexp.Compile("^(?:" + v + ")$")
+	if err != nil {
+		return nil, err
+	}
+
+	parsed, err := syntax.Parse(v, syntax.Perl)
```

Comment:
Why does this say Perl? We're not using PCRE.

Reply:
Even before this PR we compiles the regex with `regexp.Compile()` which internally does:
```
func Compile(expr string) (*Regexp, error) {
	return compile(expr, syntax.Perl, false)
}
```